src/components/Pagination.ts

```diff
@@ -0,0 +1,169 @@
+import { Component, createElement } from "react";
+import * as classNames from "classnames";
+
+import { MendixButton } from "./MendixButton";
+
+export interface PaginationProps {
+    maxPageSize: number;
+    offSet: number;
+    onClickAction: (offSet: number) => void;
+    showPageButton?: boolean;
+    setMessageStatus: (currentOffSet: number, offSet: number, maxPageSize: number) => string;
+}
+
+interface PaginationState {
+    currentOffSet: number;
+    isVisible?: boolean;
+    statusMessage: string;
+    previousIsDisabled: boolean;
+    nextIsDisabled: boolean;
+}
+
+export class Pagination extends Component<PaginationProps, PaginationState> {
+    constructor(props: PaginationProps) {
+        super(props);
+
+        this.state = {
+            currentOffSet: 0,
+            isVisible: this.props.showPageButton,
+            nextIsDisabled: false,
+            previousIsDisabled: true,
+            statusMessage: ""
+
+        };
+
+        this.firstPageClickAction = this.firstPageClickAction.bind(this);
+        this.lastPageClickAction = this.lastPageClickAction.bind(this);
+        this.nextPageClickAction = this.nextPageClickAction.bind(this);
+        this.previousPageClickAction = this.previousPageClickAction.bind(this);
+        this.callOnClickAction = this.callOnClickAction.bind(this);
+    }
+
+    render() {
+        return createElement("div", {
+                className: classNames("page-button",
+                    `visible: ${this.state.isVisible ? "visible" : "hidden"}`
+                )
+            },
+            createElement(MendixButton, {
+                buttonType: "first",
+                glyphIcon: "step-backward",
+                isDisabled: this.state.previousIsDisabled,
+                onClickAction: this.firstPageClickAction
+            }),
+            createElement(MendixButton, {
```

Comment:
@edwinmp ,Actually there is no repetition at all. Creating a function does not make any improvement because even if it is function.  The  number of arguments passed is still the same and the number of time the function will be called is still the same. I have opted not to inline and go for visibility and and here see below:

Reply:
render() {
        return createElement("div", {
                className: classNames("pagination",
                    `visible: ${this.state.isVisible ? "visible" : "hidden"}`
                )
            },
            this.createFirstButton(),
            this.createPreviousButton(),
            createElement("span", { className: "paging-status" },
                this.state.statusMessage
            ),
            this.createNextButton(),
            this.createLastButton()
        );
    }